pegasus/router.rb

```diff
@@ -383,6 +384,36 @@ def resolve_template(subdir, extnames, uri)
       nil
     end
 
+    # Scans the filesystem and finds all documents served by Pegasus CMS.
+    # @return [Array<Hash<Symbol, String>] An array of :site, :uri hash entries for all found documents.
+    def all_documents
+      dirs = Dir.children(content_dir).select {|file| Dir.exist?(content_dir(file))}
+      dirs.map do |site|
+        (settings.template_extnames - ['.fetch']).map do |ext|
+          site_glob = site_sub = content_dir(site, 'public')
+
+          if site == 'hourofcode.com'
+            # hourofcode.com has custom logic to include
+            # optional `/i18n` folder in its file-search path.
+            site_glob.sub!(site, "{#{site},#{site}/i18n}")
+            site_sub = /#{content_dir(site)}(\/i18n)?\/public/
+          end
+
+          Dir.glob("#{site_glob}/**/*#{ext}").map do |file|
+            # Reduce file to URI.
+            uri = file.
+              sub(site_sub, '').
+              sub(/#{File.extname(file)}$/, '').
+              sub(/\/index$/, '')
+            next if uri.start_with?('/private')
+            uri.prepend('/us') if site == 'hourofcode.com'
+            site = 'italia.code.org' if site == 'i18n.code.org'
```

Comment:
why italia?

Reply:
I can add comments to these two tweaks:

- `i18n.code.org` is currently not exposed as its own but serves as the base layer (configured in `/config.json`) for each of `CDO.partners`. In order to correctly cover (render) the public documents from this base layer, one of the partner domains needs to be chosen. I chose `italia.code.org` because on a manual check, it exposes all of the documents served by `i18n.code.org` (that is, it doesn't overlay any of them).
- Prepend `/us` to `hourofcode.com`-site documents, because of [some special hourofcode.com routing logic](https://github.com/code-dot-org/code-dot-org/blob/a58ef1aae1ba69900fbb5bbdbebc1e6a3196f1c1/pegasus/helpers/hourofcode_helpers.rb#L31-L66) that redirects to a country-prefixed path if a country isn't already provided in the path. Any country-code would work here, but `/us` is already the hardcoded default/fallback in our codebase.

Reply:
oops, the redirect-following part for the hourofcode.com routing logic didn't work as I intended and broke some of the routing behavior. I reverted that part back to prepending `/us` in #21283.